### PR TITLE
Case 20887: Move FST joint property handling to the model preparation step

### PR DESCRIPTION
**Pull request by sabrina-shanman**
https://highfidelity.manuscript.com/f/cases/20887/Move-FST-joint-association-to-Model-Preparation-Step

This PR moves some of the processing of the FST file out of FBXSerializer and into the model prep step, with a focus on joints. It also cleans up some unused properties I found while figuring out what to move.

The following FST properties now remain in FBXSerializer: "bs" (for blendshapes), "materialMap", and properties related to the rescaling of the model.

## TEST PLAN
- Verify animation for arbitrary test avatars (walking, emotes) is the same as in master, and the avatar's default resting pose is the same.
- Verify avatar packager still works

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16881/HighFidelity-Beta-PR14802-203e8e2.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16881/HighFidelity-Beta-PR14802-203e8e2.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16881/HighFidelity-Beta-Interface-PR14802-203e8e2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16881/HighFidelity-Beta-PR14802-203e8e2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16881/HighFidelity-Beta-Interface-PR14802-203e8e2.exe

File: libraries/model-baker/src/model-baker/PrepareJointsTask.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    // Get joint rotation offsets from FST file mappings
+    auto offsets = getJointRotationOffsets(mapping);
+    for (auto itr = offsets.begin(); itr != offsets.end(); itr++) {
+        QString jointName = itr.key();
+        glm::quat rotationOffset = itr.value();
+        int jointIndex = jointIndices.value(jointName) - 1;
+        if (jointIndex != -1) {
+            jointRotationOffsets.insert(jointIndex, rotationOffset);
+        }
+        qCDebug(model_baker) << "Joint Rotation Offset added to Rig._jointRotationOffsets : " << " jointName: " << jointName << " jointIndex: " << jointIndex << " rotation offset: " << rotationOffset;
```
Comment (AndrewMeadows) [inline libraries/model-baker/src/model-baker/PrepareJointsTask.cpp]: I know you're just moving existing code code from elsewhere to here, but this looks slightly incorrect:
The `qCDebug()` claims **Joint Rotation Offset added** even when not true (e.g. when `jointIndex == -1`).  For more correctness `qCDebug()` call should be moved inside the `if` block.
Once you do that: `rotationOffset` would only ever be used inside the if block, and could be moved entirely inside that scope.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16974/HighFidelity-Beta-PR14802-3d1edf4.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16974/HighFidelity-Beta-PR14802-3d1edf4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16974/HighFidelity-Beta-Interface-PR14802-3d1edf4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16974/HighFidelity-Beta-PR14802-3d1edf4.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16974/HighFidelity-Beta-Interface-PR14802-3d1edf4.exe

**Pull request merged**